crates/rattler/src/install/installer/mod.rs

```diff
@@ -680,3 +717,390 @@ async fn populate_cache(
         .await
         .map_err(|e| InstallerError::FailedToFetch(record.file_name.clone(), e))
 }
+
+/// Creates a mapping from package names to their requested spec strings.
+///
+/// This function takes a list of `MatchSpecs` and creates a mapping where:
+/// - The key is the `PackageName` from the `MatchSpec`
+/// - The value is a vector of string representations of all matching `MatchSpecs`
+///
+/// `MatchSpecs` without names are skipped.
+/// For multiple `MatchSpecs` with the same package name, all are collected.
+fn create_spec_mapping(specs: &[MatchSpec]) -> std::collections::HashMap<PackageName, Vec<String>> {
+    let mut mapping = std::collections::HashMap::new();
+
+    for spec in specs {
+        if let Some(name) = &spec.name {
+            mapping
+                .entry(name.clone())
+                .or_insert_with(Vec::new)
+                .push(spec.to_string());
+        }
+    }
+
+    mapping
+}
+
+/// Updates existing `PrefixRecord` files with their requested specs.
+///
+/// This function takes existing records that weren't modified by the transaction
+/// and updates their `requested_spec` field based on the spec mapping or clears it if requested.
+/// The updated records are then written back to disk.
+fn update_existing_records(
+    existing_records: &[PrefixRecord],
+    spec_mapping: &HashMap<PackageName, Vec<String>>,
+    prefix: &Prefix,
+) -> Result<(), InstallerError> {
+    for record in existing_records {
```

Comment:
Should we use `rayon` to parallelize this loop?

Reply:
Changed to use rayon!